spec/models/invoice_spec.rb

```diff
@@ -143,13 +151,118 @@
-          invoice.save
+          invoice.save!
+          invoice.finalized!
 
           organization_id_substring = organization.id.last(4).upcase
 
           expect(invoice.number).to eq("LAG-#{organization_id_substring}-#{Time.now.utc.strftime('%Y%m')}-016")
         end
       end
+
+      context 'with existing draft invoices that have generated sequential ids' do
+        let(:created_at) { Time.now.utc }
+
+        let(:invoice1) do
+          create(
+            :invoice,
+            customer:,
+            organization:,
+            sequential_id: 4,
+            organization_sequential_id: 14,
+            created_at:,
+            status: :draft,
+            number: "LAG-#{organization.id.last(4).upcase}-#{Time.now.utc.strftime('%Y%m')}-014"
+          )
+        end
+        let(:invoice2) do
+          create(
+            :invoice,
+            customer:,
+            organization:,
+            sequential_id: 5,
+            organization_sequential_id: 15,
+            created_at:,
+            number: "LAG-#{organization.id.last(4).upcase}-#{Time.now.utc.strftime('%Y%m')}-015"
+          )
+        end
+
+        before do
+          invoice1
+          invoice2
+        end
+
+        it 'scopes the organization_sequential_id to the organization and month' do
+          invoice.save!
+          invoice.finalized!
+
+          organization_id_substring = organization.id.last(4).upcase
+
+          expect(invoice.number).to eq("LAG-#{organization_id_substring}-#{Time.now.utc.strftime('%Y%m')}-016")
+
+          invoice1.update!(payment_due_date: invoice1.payment_due_date + 1.day)
+          invoice2.update!(payment_due_date: invoice2.payment_due_date + 1.day)
+
+          expect(invoice1.reload.number).to eq("LAG-#{organization_id_substring}-#{Time.now.utc.strftime('%Y%m')}-014")
+          expect(invoice2.reload.number).to eq("LAG-#{organization_id_substring}-#{Time.now.utc.strftime('%Y%m')}-015")
+
+          invoice1.finalized!
+
+          expect(invoice1.reload.number).to eq("LAG-#{organization_id_substring}-#{Time.now.utc.strftime('%Y%m')}-014")
+        end
+      end
+
+      context 'with existing draft invoices that does not have generated sequential ids' do
+        let(:created_at) { Time.now.utc }
+
+        let(:invoice1) do
+          create(
+            :invoice,
+            customer:,
+            organization:,
+            sequential_id: nil,
+            organization_sequential_id: 0,
+            created_at:,
+            status: :draft,
+            number: "LAG-#{organization.id.last(4).upcase}-DRAFT"
+          )
+        end
+        let(:invoice2) do
+          create(
+            :invoice,
+            customer:,
+            organization:,
+            sequential_id: 4,
+            organization_sequential_id: 14,
+            created_at:,
+            number: "LAG-#{organization.id.last(4).upcase}-#{Time.now.utc.strftime('%Y%m')}-014"
```

Comment:
Style/TrailingCommaInArguments: Put a comma after the last parameter of a multiline method call. (https://rubystyle.guide#no-trailing-params-comma)